src/traits/impls.rs

```diff
@@ -53,24 +57,26 @@ impl DoughnutApi for () {
 
 #[cfg(feature = "std")]
 impl Signing for DoughnutV0 {
-    fn sign_ed25519(&mut self, secret_key: &[u8]) -> Result<H512, SigningError> {
+    fn sign_ed25519(&mut self, secret_key: &[u8]) -> Result<Vec<u8>, SigningError> {
         sign_ed25519(&self.issuer(), secret_key, &self.payload())
```

Comment:
I think this could make bad signatures since we set the signature version after signing

Reply:
Interesting! I just saw this comment. We both thought about the same thing at the same time. Basically the root cause of the test issues was that they were checking the version before even calling this function. I guess we can address your concern here about the signature in https://centralitydev.atlassian.net/browse/IND-180

Reply:
This is still a bug. The other change would be an improvement

Reply:
It doesn't look like a bug to me. We are setting both the signature and signature version in the same scope and there is no conditional path in between. Would you elaborate more on this?
On the other side, returning a valid value for the signature version when it's invalid is a bug. If you ask me, IND-180 was the root cause of the test case issues that Ben had added. A valid value for an invalid field might even seem as correct in some scenarios and make the situation more confusing.

Reply:
`sign_ed25519(&self.issuer(), secret_key, &self.payload())` happens first using the `signature_version` at the time of calling which could be anything. We set the signature then change the `signature_version` which should invalidate the signature